src/Utils.ts

```diff
@@ -986,12 +986,129 @@ function setDummyInputDebugValue(
+    private _dummies: WeakRef<HTMLElement>[] = [];
+    domChanged?(parent: HTMLElement): void;
+
+    constructor(win: GetWindow) {
+        this._win = win;
+    }
+
+    add(dummy: HTMLElement, callback: () => void): void {
+        const dummyMap = this._dummyMap;
+        const cb = dummyMap.get(dummy);
+
+        if (!cb) {
+            this._dummies.push(new WeakRef(dummy));
+            this.domChanged = this._domChanged;
+        }
+
+        dummyMap.set(dummy, callback);
+    }
+
+    remove(dummy: HTMLElement): void {
+        const dummyMap = this._dummyMap;
+        const cb = dummyMap.get(dummy);
+
+        if (cb) {
+            dummyMap.delete(dummy);
+
+            const dummies = this._dummies;
+
+            for (let i = dummies.length; i--; ) {
+                const el = dummies[i].deref();
+
+                if (el) {
+                    if (el === dummy) {
+                        dummies.splice(i, 1);
+                        break;
+                    }
+                } else {
+                    dummies.splice(i, 1);
+                }
+            }
+
+            if (dummies.length === 0) {
+                delete this.domChanged;
+            }
+        }
+    }
+
+    dispose(): void {
+        const win = this._win?.();
+
+        if (this._offsetsTimer) {
+            win?.clearTimeout(this._offsetsTimer);
+            delete this._offsetsTimer;
+        }
+
+        if (this._domChangedTimer) {
+            win?.clearTimeout(this._domChangedTimer);
+            delete this._domChangedTimer;
+        }
+
+        this._domChangedMap = new WeakMap();
+        this._dummyMap = new WeakMap();
+        this._dummies = [];
+
+        delete this._win;
+    }
+
+    private _domChanged = (parent: HTMLElement): void => {
+        if (this._domChangedMap.has(parent)) {
+            return;
+        }
+
+        this._domChangedMap.set(parent, true);
+
+        if (this._domChangedTimer) {
+            return;
+        }
+
+        this._domChangedTimer = this._win?.().setTimeout(() => {
+            delete this._domChangedTimer;
+
+            for (const dummyRef of this._dummies) {
+                const dummy = dummyRef.deref();
+                const dummyParent = dummy?.parentElement;
+
+                if (
+                    dummy &&
+                    (!dummyParent || this._domChangedMap.has(dummyParent))
+                ) {
+                    this._dummyMap.get(dummy)?.();
+                }
+            }
+
+            this._domChangedMap = new WeakMap();
+        }, 100);
+    };
+
+    updateOffsets(callback: () => void): void {
+        this._offsetsQueue.push(callback);
+
+        if (!this._offsetsTimer) {
+            this._offsetsTimer = this._win?.().setTimeout(() => {
+                delete this._offsetsTimer;
+
+                this._offsetsQueue.forEach((callback) => callback());
+                this._offsetsQueue = [];
+            }, 100);
+        }
+    }
```

Comment:
Well, if we speak about consistency, it would require to change quite a number of places. Because to that day I was mostly everywhere doing `if (!timer) { doThings(); }` rather than `if (timer) { return; } doThings();`. Because it's better for the bundle size. @bsunderhus has asked me to change these in _domChanged for more readability.

I suppose if we really want to be consistent about it, we can try to change it everywhere (maybe even try some lint rule, if possible). But we need to decide what is more important — this kind of consistency or a bundle size. Perhaps the JS minifiers can adjust these cases automatically, makes sense to check.

Reply:
I don't mind which one you do, you can make `_domChanged` the same as what you have here, it's just a bit strange seeing both conventions in the same class